Cython/Compiler/Symtab.py

```diff
@@ -470,16 +470,21 @@ def declare(self, name, cname, type, pos, visibility, shadow = 0, is_type = 0, c
             warning(pos, "'%s' is a reserved name in C." % cname, -1)
 
         entries = self.entries
+        old_index = -1
         if name and name in entries and not shadow:
             old_entry = entries[name]
 
             # Reject redeclared C++ functions only if they have the same type signature.
             cpp_override_allowed = False
             if type.is_cfunction and old_entry.type.is_cfunction and self.is_cpp():
-                for alt_entry in old_entry.all_alternatives():
-                    if type == alt_entry.type:
+                for index, alt_entry in enumerate(old_entry.all_alternatives()):
```

Comment:
This makes an assumption about the order returned by `.all_alternatives()`. The code here shouldn't rely on that.

Reply:
I understand the point. The goal here is to mark a given inherited alternative as not inherited anymore. I see two possible ways:

1. Create a new entry as the rest of the cases and override the correct alternative with this new entry
2. Update the current entry without creating a new one

I've chosen option 1, and as I need to override the correct alternative in the overloaded_alternatives array, the index pops out.
I'm perfectly OK with option 2 if you think it fits better.
That would consists in updating flags, cname and type of the correct alternative, and avoid creating a useless new entry.